app/services/SignatureService.scala

```diff
@@ -42,24 +41,25 @@ class SignatureService @Inject()() extends Logging {
 
   Security.addProvider(new BouncyCastleProvider)
 
-  def createSignatureForPass(passPath: Path,
+
+  def createSignatureForPass(passContent: List[FileAsBytes],
                              privateCertificate: String,
                              privateCertificatePassword: String,
                              appleWWDRCACertificate: String
-                            ): Boolean = {
-    val resultForCreateSignature = for {
-      signInfo <- loadSigningInformation(privateCertificate, privateCertificatePassword, appleWWDRCACertificate)
-      processableFile <- Try(new CMSProcessableFile(passPath.resolve(FileService.MANIFEST_JSON_FILE_NAME).toFile))
-      signContent <- signManifestUsingContent(processableFile, signInfo)
-      result <- Try(Files.write(passPath.resolve(SIGNATURE_FILE_NAME), signContent))
-    } yield result
-
-    resultForCreateSignature match {
-      case Success(_) => true
-      case _ =>
-        false
+                            ): FileAsBytes = {
+
+    if (passContent.nonEmpty) {
+      val resultForCreateSignature = for {
+        signInfo <- loadSigningInformation(privateCertificate, privateCertificatePassword, appleWWDRCACertificate)
+        processableFileBytes <- Try(new CMSProcessableByteArray(passContent.last.content))
```

Comment:
A list is ordered so this should always return what is expected. Do we have some spec's around this? I.e. have we tested the 'try' and validated the outcome?

Reply:
I will double check this now, I only tested if the entire function were to get a invalid pass content or a valid pass content but not the specifically the `processableFileBytes <- Try(new CMSProcessableByteArray(passContent.last.content))
`

Reply:
Update: testing if SignatureService is able to generate a signature requires the correct private certificate key which I do not think I can stub or mock (if I am wrong please correct me). I think this would require testing in the local environments for validity if I cannot get a spec for this.